### PR TITLE
fix(issues): Add partition key for status change kafka msg

### PR DESCRIPTION
**Pull request by aliu39**
Reopened from https://github.com/getsentry/sentry/pull/71620
Previously status change had no key. There are no order guarantees across partitions, resulting in status changes being processed before the first occurrence/group was created.

### CODE REVIEW

**Review by JoshFerge — APPROVED** [review]

File: tests/sentry/issues/test_producer.py [hunk 1]
```diff
@@ -376,3 +376,81 @@ def test_generate_status_changes_id(self) -> None:
         )
         assert status_change_1.id
         assert status_change_1.id != status_change_2.id
+
+    @patch(
```
Comment (aliu3ntry) [inline tests/sentry/issues/test_producer.py:380]: These are copied from the occurrence tests

**Review by wedamija — APPROVED** [review]
This works fine, any reason we can't use the value from the payload though?

**Comment by aliu3ntry** [comment]
> This works fine, any reason we can't use the value from the payload though?

There's no guarantee from the `_prepare` functions that `payload_data` contains a fingerprint, or a non-empty fingerprint. We could enforce this, but we'd have to rewrite the tests that still produce when the fingerprint is missing. 
I'm not sure if this (no fingerprint -> fail) is the behavior issues wants

**Comment by aliu3ntry** [comment]
CI was failing in s4s, want to be around when this goes out so will try again tmrw

**Pull request merged**